packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -106,125 +36,26 @@ function calculateContent(trace: Trace, span: Span, allSpans: Span[], resultValu
-          tempSelf = earliestLongerAsParent.relativeStartTime - span.relativeStartTime;
-          for (let i = 0; i < allChildrenWithout.length; i++) {
-            tempSelf -= allChildrenWithout[i].duration;
-          }
-        } else {
-          const overlayOnly = _.difference(allOverlay, longerAsParentSpan);
-          const allChildrenWithout = _.difference(allChildren, longerAsParentSpan);
-          const earliestLongerAsParent = _.minBy(longerAsParentSpan, function calc(a) {
-            return a.relativeStartTime;
-          });
-
-          // overlay between longerAsParent and overlayOnly
-          const overlayWithout = [];
-          for (let i = 0; i < overlayOnly.length; i++) {
-            if (!earliestLongerAsParent.relativeStartTime <= overlayOnly[i].relativeStartTime) {
-              overlayWithout.push(overlayOnly[i]);
-            }
-          }
-          for (let i = 0; i < overlayWithout.length; i++) {
-            if (
-              overlayWithout[i].relativeStartTime + overlayWithout[i].duration >
-              earliestLongerAsParent.relativeStartTime
-            ) {
-              overlayWithout[i].duration -=
-                overlayWithout[i].relativeStartTime +
-                overlayWithout[i].duration -
-                earliestLongerAsParent.relativeStartTime;
-              if (overlayWithout[i].duration < 0) {
-                overlayWithout[i].duration = 0;
-              }
-            }
-          }
-
-          tempSelf = onlyOverlay(overlayWithout, allChildrenWithout, tempSelf, span);
-          const diff = span.relativeStartTime + span.duration - earliestLongerAsParent.relativeStartTime;
-          tempSelf = Math.max(0, tempSelf - diff);
-        }
-      } else if (longerAsParent) {
-        // span is longer as Parent
-        tempSelf = longerAsParentSpan[0].relativeStartTime - span.relativeStartTime;
-        for (let i = 0; i < allChildren.length; i++) {
-          if (allChildren[i].spanID !== longerAsParentSpan[0].spanID) {
-            tempSelf -= allChildren[i].duration;
-          }
-        }
-      } else {
-        // Overlay
-        tempSelf = onlyOverlay(allOverlay, allChildren, tempSelf, span);
-      }
-    }
-    // no children
+    const spanRange = new DRange(span.startTime, span.startTime + span.duration);
+    // Filter for CHILD_OF we don't want to calculate FOLLOWS_FROM
+    const children = allSpans.filter(
+      each =>
+        span.childSpanIds.includes(each.spanID) &&
+        each.references[0].spanID === span.spanID &&
+        each.references[0].refType === 'CHILD_OF'
+    );
+    children.forEach(child => {
+      // A DRange(start,end) builds interval as whole numbers.
+      // So when subtarcting a DRange directly do (+1 , -1) to (start,end) otherwise
+      // for each subtaracting of a DRange it will take extra 2 microseconds
+      spanRange.subtract(child.startTime + 1, child.startTime + child.duration - 1);
```

Comment:
what if `child.duration == 1`?  It will be invalid drange

Reply:
yes, I think it is an exceptional case.
May be we should write an exceptional code in if block `if(span.duaration <=1)`
```
const n;
if(child.duration <=1) n += 1;
else{
   spanRange.subtract(child.startTime + 1, child.startTime + child.duration - 1);
}
```

finally we should subtract n from selfTime
may be at here temself = tempself + selfTime - n
https://github.com/jaegertracing/jaeger-ui/blob/53dd0df12b554313956876d0e9aec690e24990e1/packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx#L58

Reply:
> finally we should subtract n from selfTime

this would only be correct if no other child overlaps with that d=1 child.

Maybe another way to do this is to fool drange to support open intervals. For example, if we multiply all values by 10 and create the ranges as `[10*startTime, 10 * (startTime+duration) - 1]`. So what would be `[10, 20]` in the current code would become `[100, 199]`. Then subtracting ranges would always result in the upper bound of left remaining segment to end with 9 and the lower bound of the next segment to be multiple of 10, i.e. we'd never have improper overlaps.

Reply:
Yes that works!!